upstairs/src/lib.rs

```diff
@@ -6192,6 +6250,30 @@ impl Guest {
         Ok(self.send(wio))
     }
 
+    // Guest does support write_unwritten
+    pub fn write_unwritten(
+        &self,
+        offset: Block,
+        data: Bytes,
+    ) -> Result<BlockReqWaiter, CrucibleError> {
+        if !self.is_active() {
+            return Err(CrucibleError::UpstairsInactive);
+        }
+
```

Comment:
With the new read-only Upstairs flag, you'll have to check for that here and return `ModifyingReadOnlyRegion`

Reply:
Yess. well.
At this layer, the Guest does not know about the Upstairs and if it is read only or not.  The check for that would have to come at a higher level. 

We (you added?) do check in `submit_write()` for RO status, but I don't think we check anywhere in the Guest/BlockIO/DownstairsIO path for RO status for even writes yet, right?

Reply:
I thought this was inside Upstairs for some reason, you're totally right. Disregard this comment.